src/spi.rs

```diff
@@ -4,32 +4,108 @@ use core::ptr;
 
 use crate::hal::spi::FullDuplex;
 pub use crate::hal::spi::{Mode, Phase, Polarity};
+#[cfg(any(
+    feature = "stm32f302xd",
+    feature = "stm32f302xe",
+    feature = "stm32f303xd",
+    feature = "stm32f303xe",
+    feature = "stm32f398",
+))]
+use crate::stm32::SPI4;
 use crate::stm32::{SPI1, SPI2, SPI3};
 use nb;
 
+#[cfg(any(feature = "stm32f373", feature = "stm32f378"))]
+use crate::gpio::gpioa::{PA1, PA10, PA12, PA13, PA2, PA3, PA8, PA9};
+#[cfg(any(
+    feature = "stm32f302x6",
```

Comment:
If this feature is not available, I would add it, as we do now have a reason to differentiate between those features / devices.